packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1646,24 +1761,60 @@ export class Vpc extends VpcBase {
         // For reserved azs, do not create any resources
         return;
       }
+      let subnetProps: SubnetProps;
+      if (this.vpcProtocol === VpcProtocol.IPV4_ONLY) {
```

Comment:
Don't do a top-level if on the VPC protocol.

Do a per-property calculation that can depend on the VpcProtocol instead (if relevant).